### PR TITLE
Fixing range for primary edit

### PR DESCRIPTION
**Pull request by navya9singh**
Fixes https://github.com/microsoft/TypeScript/issues/58936 
Due to `getText()` and not `getFullText()`, new line characters were getting ignored when trying to perform a temporary file update which later made the paste at the wrong positions.

### CODE REVIEW

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]
There is another call to `getText()` below that is incorrectly used. It's arguable whether or not that should just reuse `originalText`, but I honestly am not familiar enough with the code.

But something must be wrong when copying/pasting trivia over, and the new test must not cover it.

**Comment by navya9singh** [comment]
> There is another call to `getText()` below that is incorrectly used. It's arguable whether or not that should just reuse `originalText`, but I honestly am not familiar enough with the code.
> 
> But something must be wrong when copying/pasting trivia over, and the new test must not cover it.

I haven't found a case where the next `getText()` would be a problem, but yes I think it's better to have that instead. 
This part of the code takes the targetFile, and adds the new, 'pasted' text to it, then performs the callback function to deal with imports. In the finally block, it adds the original text to the target file which was there before the paste took place.

**Review by sheetalkamat — APPROVED** [review]

File: src/server/project.ts [hunk 1]
```diff
@@ -2318,15 +2318,15 @@ export abstract class Project implements LanguageServiceHost, ModuleResolutionHo
...
+        const originalText = Debug.checkDefined(rootSourceFile.getFullText());
 
         this.getScriptInfo(rootFile)?.editContent(0, originalText.length, updatedText);
         this.updateGraph();
         try {
             cb(this.program!, originalProgram, (this.program?.getSourceFile(rootFile))!);
         }
         finally {
-            this.getScriptInfo(rootFile)?.editContent(0, this.program!.getSourceFile(rootFile)!.getText().length, originalText);
+            this.getScriptInfo(rootFile)?.editContent(0, this.program!.getSourceFile(rootFile)!.getFullText().length, originalText);
```
Comment (sheetalkamat) [inline src/server/project.ts:2329]:
```suggestion
 this.getScriptInfo(rootFile)?.editContent(0, updatedText.length, originalText);
```

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Review by sheetalkamat — APPROVED** [review]

File: src/server/project.ts [hunk 2]
```diff
@@ -2318,15 +2318,15 @@ export abstract class Project implements LanguageServiceHost, ModuleResolutionHo
...
+        const originalText = Debug.checkDefined(rootSourceFile.getFullText());
 
         this.getScriptInfo(rootFile)?.editContent(0, originalText.length, updatedText);
         this.updateGraph();
         try {
             cb(this.program!, originalProgram, (this.program?.getSourceFile(rootFile))!);
         }
         finally {
-            this.getScriptInfo(rootFile)?.editContent(0, this.program!.getSourceFile(rootFile)!.getText().length, originalText);
+            this.getScriptInfo(rootFile)?.editContent(0, updatedText.length, originalText);
```
Comment (DanielRosenwasser) [inline src/server/project.ts:2329]: Hmm... looking at it again, is it really safe to assume that `cb` doesn't itself change the contents of the file? I'm sure the way we use this function is used in our codebase, it's fine, but I just want to check. What do you think @sheetalkamat?

Comment (sheetalkamat) [reply src/server/project.ts:2329]: yes we should assume cb doesnt change the contents of file since file is never "muted" without change in script info

Comment (DanielRosenwasser) [reply src/server/project.ts:2329]: (assuming you mean mutated)

What I meant was that `cb` could itself run `editContent` somehow. If that happens, the `ScriptInfo` object may not actually contain the same text anymore.

I'm okay with assuming that for now, I just wanted to run it by you.

**Review by DanielRosenwasser — APPROVED** [review]

File: src/server/project.ts (hunk 2 above)
Comment (DanielRosenwasser) [reply src/server/project.ts:2329]: The alternative is just to replace the text contents with the script info's own text contents. It should all be available anyway.

Comment (sheetalkamat) [reply src/server/project.ts:2329]: Yes I did mean mutated ☺️ Currently editing script info from callback is not possible as it’s just trying to calculate import fixes.. when that changes we can revisit this  ..

**Pull request merged**